### DISCUSSION TITLE
How pointers should be verifies as null?

### DISCUSSION THREAD

**Discussion by IsraelBuitronD** [Snaipe/Criterion#448] [Q&A] [answered]
Given version `2.4.1` and new assertion macros, how should pointers be verified as null or as not null?

For example, let us:

```c
int *mypointer = (int*)malloc(2*sizeof(int));
free(mypointer);
```

How should assertion be written to verify `mypointer` is effectively freed and how to verify is not null (if it is expected)?

Before `2.4.1` we use to have `cr_assert_null(mypointer)` or `cr_assert_not_null(mypointer)`.

Thank you.

**Comment by Snaipe** [comment, marked as answer]
The `zero` criterion is what you want: https://criterion.readthedocs.io/en/v2.4.1/assert.html#c.zero

```
cr_assert(zero(ptr, mypointer));
cr_assert(not(zero(ptr, mypointer)));
```

More examples on how to use the new assertions can be seen in the [asserts.c sample](https://github.com/Snaipe/Criterion/blob/bleeding/samples/asserts.c)
